magento/magento2
# GraphQL - Return shipping total including tax in shipping method resolver

### Description (*)
Return shipping total including tax for selected shipping method resolver. Vanilla Magento will always display the shipping price including tax within cart price summary, regardless of `tax/calculation/shipping_includes_tax` value. This logic is not mirrored in the graphql resolvers, however, as the base shipping amount is returned.

**Vanilla Magento checkout with `tax/calculation/shipping_includes_tax` set to `1`:**
![image](https://user-images.githubusercontent.com/22290866/164221372-14d3042d-70b7-43c0-a8a4-ea71e67b27de.png)

**GraphQL cart query for shipping totals with `tax/calculation/shipping_includes_tax` set to `1`:**
```graphql
query getCartDetails($cartId: String!) {
    cart(cart_id: $cartId) {
        shipping_addresses {
            selected_shipping_method {
                amount {
                    value
                    currency
                }
                carrier_code
                carrier_title
                method_code
                method_title
            }
        }
    }
}
```
```json
{
  "data": {
    "cart": {
      "shipping_addresses": [
        {
          "selected_shipping_method": {
            "amount": {
              "value": 8.33,
              "currency": "GBP"
            },
            "carrier_code": "testshipping",
            "carrier_title": "Shipping",
            "method_code": "testshipping",
            "method_title": "Shipping"
          }
        }
      ]
    }
  }
}
```

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)
- Setup latest version of M2.4
- Set `tax/calculation/shipping_includes_tax` to `1` (or manually define shipping price to "Include Tax" via admin panel)
- Configure standard tax class/rates of 20% for taxable goods
- Create test product, assign to taxable goods tax class
- Set "Flat Rate" shipping method "Enabled" field to "Yes" via admin panel
- Set "Flat Rate" shipping method "Type" field to "Per Order" via admin panel
- Set "Flat Rate" shipping method "Price" field to "10" via admin panel
- Add test product to cart
- Navigate to checkout, fill in shipping address data, select flat rate shipping method
- Confirm order summary in checkout and cart lists shipping price as 10
- Run example graphql query noted in ticket description on the same cart
- Confirm that value returned is **not** the same as the value returned in checkout/cart summary

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#36876: Return shipping total including tax
